### PR TITLE
Use alias for LNet index 0

### PR DESCRIPTION
**Pull request by matthew-richerson**
If the MGS address specified for an external MGS uses an LNet that ends with index "0", trim the "0" off the address. Lustre aliases the 0 index with no index, and it uses no index in the "mount" command output.

For example:
10.1.1.113@tcp0 -> 10.1.1.113@tcp

### CODE REVIEW

File: internal/controller/nnf_workflow_controller_helpers.go [hunk 1]
```diff
@@ -694,6 +695,55 @@ func (r *NnfWorkflowReconciler) createNnfStorage(ctx context.Context, workflow *
 	return nnfStorage, nil
 }
 
+func (r *NnfWorkflowReconciler) trimMgsAddress(mgsAddress string) string {
```
Comment (bdevcich) [inline internal/controller/nnf_workflow_controller_helpers.go:698]: You're not using `r` anywhere, so this can be a static function.

Comment (bdevcich) [inline internal/controller/nnf_workflow_controller_helpers.go:698]: Also, could you explain what this is doing in a comment?

File: internal/controller/nnf_workflow_controller_helpers.go [hunk 2]
```diff
@@ -694,6 +695,65 @@ func (r *NnfWorkflowReconciler) createNnfStorage(ctx context.Context, workflow *
...
+// trimMgsAddress takes an MGS address as given in the NnfStorageProfile ExternalMgs field
+// and removes the "0" for any addresses using LNet 0. This is needed because Lustre trims the
+// "0" off internally, so the MGS address given in the "mount" command output is trimmed. We need
+// the "mount" command output to match our view of the MGS address so we can verify if the file system
+// is mounted correctly.
+// Examples:
+// 25@kfi0:26@kfi0 -> 25@kfi:26@kfi
+// 25@kfi10 -> 25@kfi10
+// 10.1.1.113@tcp0 -> 10.1.1.113@tcp
+// 25@kfi0,25@kfi1:26@kfi0,26@kfi1 -> 25@kfi,25@kfi1:26@kfi,26@kfi1
```
Comment (bdevcich) [inline internal/controller/nnf_workflow_controller_helpers.go:707]: Thanks, this helps!

**Comment by roehrich-hpe** [comment]
I had to spend some time playing with that. It's not so obvious.

I think you can replace the whole thing with the following:

```
package main

import (
	"fmt"
	"regexp"
)

func main() {
	m := "joe@tcp0,jack@tcp:betty@tcp1,john@tcp0:jenny@tcp"
	fmt.Println("Begin:", m)

	re := regexp.MustCompile("(@[A-Za-z]+)0")
	x := re.ReplaceAllString(m, "$1")
	fmt.Println("End:", x)
}
```

Yields:
```
Begin: joe@tcp0,jack@tcp:betty@tcp1,john@tcp0:jenny@tcp
End: joe@tcp,jack@tcp:betty@tcp1,john@tcp:jenny@tcp
```

**Review by bdevcich — APPROVED** [review]

**Comment by matthew-richerson** [comment]
> I had to spend some time playing with that. It's not so obvious.
> 
> I think you can replace the whole thing with the following:
> 
> ```
> package main
> 
> import (
> 	"fmt"
> 	"regexp"
> )
> 
> func main() {
> 	m := "joe@tcp0,jack@tcp:betty@tcp1,john@tcp0:jenny@tcp"
> 	fmt.Println("Begin:", m)
> 
> 	re := regexp.MustCompile("(@[A-Za-z]+)0")
> 	x := re.ReplaceAllString(m, "$1")
> 	fmt.Println("End:", x)
> }
> ```
> 
> Yields:
> 
> ```
> Begin: joe@tcp0,jack@tcp:betty@tcp1,john@tcp0:jenny@tcp
> End: joe@tcp,jack@tcp:betty@tcp1,john@tcp:jenny@tcp
> ```

Your regex skills are better than mine. I'll try that out - it's much more straight forward.

File: internal/controller/nnf_workflow_controller_helpers.go [hunk 3]
```diff
@@ -694,6 +705,17 @@ func (r *NnfWorkflowReconciler) createNnfStorage(ctx context.Context, workflow *
 	return nnfStorage, nil
 }
 
+// trimMgsAddress takes an MGS address as given in the NnfStorageProfile ExternalMgs field
```
Comment (roehrich-hpe) [inline internal/controller/nnf_workflow_controller_helpers.go:708]: Remove this comment block. You have it above..

**Review by roehrich-hpe — APPROVED** [review]

File: internal/controller/nnf_workflow_controller_helpers.go (hunk 3 above)
Comment (matthew-richerson) [reply internal/controller/nnf_workflow_controller_helpers.go:708]: I forgot you guys get updates on my commits in this PR. It's not ready yet. I'll re-open after it's been tested

**Review by roehrich-hpe — APPROVED** [review]

File: internal/controller/nnf_workflow_controller_helpers.go [hunk 4]
```diff
@@ -620,7 +621,17 @@ func (r *NnfWorkflowReconciler) createNnfStorage(ctx context.Context, workflow *
...
+					// addresses using LNet 0. This is needed because Lustre trims the "0" off internally, so the MGS address
+					// given in the "mount" command output is trimmed. We need the "mount" command output to match our view of
+					// the MGS address so we can verify if the file system is mounted correctly.
+					// Examples:
+					// 25@kfi0:26@kfi0 -> 25@kfi:26@kfi
+					// 25@o2ib10 -> 25@o2ib10
+					// 10.1.1.113@tcp0 -> 10.1.1.113@tcp
+					// 25@kfi0,25@kfi1:26@kfi0,26@kfi1 -> 25@kfi,25@kfi1:26@kfi,26@kfi1
+					re := regexp.MustCompile("(@[A-Za-z0-9]+[A-Za-z])0")
+					mgsNid = re.ReplaceAllString(nnfStorageProfile.Data.LustreStorage.ExternalMGS, "$1")
```
Comment (bdevcich) [inline internal/controller/nnf_workflow_controller_helpers.go:634]: This is quite the improvement :)

**Review by bdevcich — APPROVED** [review]

**Pull request merged**